xblock/fields.py

```diff
@@ -838,31 +842,27 @@ def from_json(self, value):
         """
         Parse the date from an ISO-formatted date string, or None.
         """
-        if isinstance(value, basestring):
+        if value is None:
+            return None
 
+        if isinstance(value, basestring):
             # Parser interprets empty string as now by default
             if value == "":
                 return None
-
             try:
-                parsed_date = dateutil.parser.parse(value)
+                value = dateutil.parser.parse(value)
             except (TypeError, ValueError):
                 raise ValueError("Could not parse {} as a date".format(value))
 
-            if parsed_date.tzinfo is not None:  # pylint: disable=maybe-no-member
-                parsed_date.astimezone(pytz.utc)  # pylint: disable=maybe-no-member
-            else:
-                parsed_date = parsed_date.replace(tzinfo=pytz.utc)  # pylint: disable=maybe-no-member
-
-            return parsed_date
-
-        if value is None:
-            return None
-
-        if isinstance(value, datetime.datetime):
-            return value
+        if not isinstance(value, datetime.datetime):
+            raise TypeError(
+                "Value should be loaded from a string, a datetime object or None, not {}".format(type(value))
+            )
 
-        raise TypeError("Value should be loaded from a string, not {}".format(type(value)))
+        if value.tzinfo is not None:  # pylint: disable=maybe-no-member
+            return value.astimezone(pytz.utc)  # pylint: disable=maybe-no-member
```

Comment:
I also ran these XBlock changes through the edx-platform test suite today - and they passed.